message_ix_models/tests/model/water/data/test_water_for_ppl.py

```diff
@@ -213,93 +213,51 @@ def test_non_cooling_tec(request, test_context):
     )
 
 
-# Mock function for scen.par
-class MockScenario:
-    def par(
-        self,
-        param: Literal["bound_activity_lo", "bound_new_capacity_lo"],
-        filters: dict,
-    ) -> pd.DataFrame:
-        year_type = "year_act" if param == "bound_activity_lo" else "year_vtg"
-
-        return pd.DataFrame(
-            {
-                "node_loc": ["R12_AFR", "R12_AFR", "R12_AFR"],
-                "technology": ["coal_ppl", "coal_ppl", "coal_ppl"],
-                year_type: [2030, 2040, 2050],
-                "value": [15, 150, 2000],
-            }
-        )
-
-
 @pytest.mark.parametrize(
-    "constraint_type, year_type",
-    [("activity", "year_act"), ("new_capacity", "year_vtg")],
+    "param_name, cap_fact_parent, expected_values",
+    [
+        (
+            "historical_activity",
+            None,
+            [100 * 0.5, 150 * 1.2],
+        ),  # Only apply hold_cost multipliers
+        (
+            "historical_new_capacity",
+            pd.DataFrame(
+                {
+                    "node_loc": ["R1", "R2"],
+                    "technology": ["TechA", "TechB"],
+                    "cap_fact": [0.9, 0.9],
+                }
+            ),
+            [100 * 0.5 * 0.9 * 1.2, 150 * 1.2 * 0.9 * 1.2],
+        ),  # Apply capacity factors
+    ],
 )
-def test_relax_growth_constraint(constraint_type, year_type):
-    # Sample data for g_lo
-    g_up = pd.DataFrame(
-        {
-            "node_loc": ["R12_AFR", "R12_AFR", "R12_AFR", "R12_AFR"],
-            "technology": [
-                "coal_ppl__ot_fresh",
-                "coal_ppl__ot_fresh",
-                "coal_ppl__ot_fresh",
-                "gas_ppl__ot_fresh",
-            ],
-            "year_act": [2030, 2040, 2050, 2030],
-            "time": ["year", "year", "year", "year"],
-            "value": [-0.05, -0.05, -0.05, -0.05],
-            "unit": ["%", "%", "%", "%"],
-        }
-    )
-
-    # Sample data for ref_hist
-    ref_hist = pd.DataFrame(
+def test_apply_act_cap_multiplier(param_name, cap_fact_parent, expected_values):
```

Comment:
Did you know? One can even type hint tests and it will make them safer, too :)
Based on your parametrization above, this should be

```suggestion
def test_apply_act_cap_multiplier(param_name: str, cap_fact_parent: Optional[pd.DataFrame], expected_values: list[float]) -> None:
```

(where you might have to import `Optional` from typing).